{"kind":"symbol","topicSections":[{"generated":true,"identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/init(errorCode:context:)"],"anchor":"Initializers","title":"Initializers"},{"title":"Instance Properties","generated":true,"anchor":"Instance-Properties","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/context","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/errorCode"]},{"title":"Type Properties","anchor":"Type-Properties","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/internalFailure","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/serviceUnavailable"],"generated":true},{"generated":true,"identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/AWSErrorType-Implementations","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/CustomStringConvertible-Implementations","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/Equatable-Implementations"],"title":"Default Implementations","anchor":"Default-Implementations"}],"schemaVersion":{"patch":0,"major":0,"minor":3},"variants":[{"paths":["\/documentation\/sotocore\/awsservererror"],"traits":[{"interfaceLanguage":"swift"}]}],"sections":[],"metadata":{"navigatorTitle":[{"kind":"identifier","text":"AWSServerError"}],"title":"AWSServerError","fragments":[{"kind":"keyword","text":"struct"},{"text":" ","kind":"text"},{"kind":"identifier","text":"AWSServerError"}],"symbolKind":"struct","modules":[{"name":"SotoCore"}],"externalID":"s:8SotoCore14AWSServerErrorV","role":"symbol","roleHeading":"Structure"},"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"text":"AWSServerError","kind":"identifier"}],"languages":["swift"],"platforms":["macOS"]}]}],"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError"},"hierarchy":{"paths":[["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore"]]},"relationshipsSections":[{"type":"conformsTo","title":"Conforms To","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType","doc:\/\/codes.soto.soto-core\/s23CustomStringConvertibleP","doc:\/\/codes.soto.soto-core\/SQ","doc:\/\/codes.soto.soto-core\/s5ErrorP","doc:\/\/codes.soto.soto-core\/s8SendableP"],"kind":"relationships"}],"seeAlsoSections":[{"title":"Errors","identifiers":["doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorContext","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSClientError","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSRawError","doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/HeaderDecodingError"],"generated":true,"anchor":"Errors"}],"references":{"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/CustomStringConvertible-Implementations":{"kind":"article","role":"collectionGroup","abstract":[],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/CustomStringConvertible-Implementations","title":"CustomStringConvertible Implementations","url":"\/documentation\/sotocore\/awsservererror\/customstringconvertible-implementations","type":"topic"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSErrorType":{"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorType","role":"symbol","type":"topic","fragments":[{"text":"protocol","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSErrorType"}],"abstract":[{"text":"Standard Error type returned by Soto.","type":"text"}],"kind":"symbol","navigatorTitle":[{"kind":"identifier","text":"AWSErrorType"}],"title":"AWSErrorType","url":"\/documentation\/sotocore\/awserrortype"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/Equatable-Implementations":{"type":"topic","url":"\/documentation\/sotocore\/awsservererror\/equatable-implementations","abstract":[],"title":"Equatable Implementations","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/Equatable-Implementations","kind":"article","role":"collectionGroup"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/internalFailure":{"fragments":[{"text":"static","kind":"keyword"},{"kind":"text","text":" "},{"kind":"keyword","text":"var"},{"text":" ","kind":"text"},{"text":"internalFailure","kind":"identifier"},{"kind":"text","text":": "},{"kind":"typeIdentifier","preciseIdentifier":"s:8SotoCore14AWSServerErrorV","text":"AWSServerError"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/internalFailure","title":"internalFailure","role":"symbol","abstract":[],"url":"\/documentation\/sotocore\/awsservererror\/internalfailure","kind":"symbol","type":"topic"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/context":{"url":"\/documentation\/sotocore\/awsservererror\/context","kind":"symbol","title":"context","abstract":[{"type":"text","text":"additional context information related to the error"}],"fragments":[{"kind":"keyword","text":"let"},{"text":" ","kind":"text"},{"text":"context","kind":"identifier"},{"text":": ","kind":"text"},{"kind":"typeIdentifier","text":"AWSErrorContext","preciseIdentifier":"s:8SotoCore15AWSErrorContextV"},{"text":"?","kind":"text"}],"role":"symbol","type":"topic","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/context"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/init(errorCode:context:)":{"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/init(errorCode:context:)","role":"symbol","fragments":[{"text":"init","kind":"identifier"},{"kind":"text","text":"?("},{"text":"errorCode","kind":"externalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:SS","text":"String","kind":"typeIdentifier"},{"text":", ","kind":"text"},{"text":"context","kind":"externalParam"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:8SotoCore15AWSErrorContextV","text":"AWSErrorContext","kind":"typeIdentifier"},{"kind":"text","text":")"}],"title":"init(errorCode:context:)","url":"\/documentation\/sotocore\/awsservererror\/init(errorcode:context:)","type":"topic","abstract":[{"type":"text","text":"initialize AWSServerError"}],"kind":"symbol"},"doc://codes.soto.soto-core/documentation/SotoCore":{"type":"topic","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore","kind":"symbol","url":"\/documentation\/sotocore","abstract":[{"type":"text","text":"The core framework for Soto, a Swift SDK for AWS"}],"role":"collection","title":"SotoCore"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError":{"title":"AWSServerError","kind":"symbol","abstract":[],"fragments":[{"text":"struct","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSServerError"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError","type":"topic","role":"symbol","navigatorTitle":[{"text":"AWSServerError","kind":"identifier"}],"url":"\/documentation\/sotocore\/awsservererror"},"doc://codes.soto.soto-core/SQ":{"identifier":"doc:\/\/codes.soto.soto-core\/SQ","type":"unresolvable","title":"Swift.Equatable"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSRawError":{"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSRawError","role":"symbol","type":"topic","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSRawError"}],"abstract":[{"type":"text","text":"Raw unprocessed error."}],"kind":"symbol","navigatorTitle":[{"kind":"identifier","text":"AWSRawError"}],"title":"AWSRawError","url":"\/documentation\/sotocore\/awsrawerror"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSClientError":{"kind":"symbol","abstract":[],"type":"topic","title":"AWSClientError","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSClientError","fragments":[{"text":"struct","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSClientError"}],"url":"\/documentation\/sotocore\/awsclienterror","role":"symbol","navigatorTitle":[{"kind":"identifier","text":"AWSClientError"}]},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/errorCode":{"type":"topic","abstract":[{"text":"return error code string","type":"text"}],"url":"\/documentation\/sotocore\/awsservererror\/errorcode","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/errorCode","title":"errorCode","fragments":[{"text":"var","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"errorCode"},{"text":": ","kind":"text"},{"text":"String","kind":"typeIdentifier","preciseIdentifier":"s:SS"}],"kind":"symbol","role":"symbol"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSResponseError":{"title":"AWSResponseError","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"AWSResponseError"}],"abstract":[{"type":"text","text":"Response Error type returned by Soto if the error code is unrecognised"}],"type":"topic","kind":"symbol","url":"\/documentation\/sotocore\/awsresponseerror","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSResponseError","role":"symbol","navigatorTitle":[{"kind":"identifier","text":"AWSResponseError"}]},"doc://codes.soto.soto-core/s8SendableP":{"identifier":"doc:\/\/codes.soto.soto-core\/s8SendableP","type":"unresolvable","title":"Swift.Sendable"},"doc://codes.soto.soto-core/s23CustomStringConvertibleP":{"identifier":"doc:\/\/codes.soto.soto-core\/s23CustomStringConvertibleP","type":"unresolvable","title":"Swift.CustomStringConvertible"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSErrorContext":{"type":"topic","url":"\/documentation\/sotocore\/awserrorcontext","title":"AWSErrorContext","kind":"symbol","role":"symbol","navigatorTitle":[{"kind":"identifier","text":"AWSErrorContext"}],"fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"text":"AWSErrorContext","kind":"identifier"}],"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSErrorContext","abstract":[{"type":"text","text":"Additional information about error"}]},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/serviceUnavailable":{"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/serviceUnavailable","role":"symbol","fragments":[{"text":"static","kind":"keyword"},{"kind":"text","text":" "},{"text":"var","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"serviceUnavailable"},{"text":": ","kind":"text"},{"text":"AWSServerError","kind":"typeIdentifier","preciseIdentifier":"s:8SotoCore14AWSServerErrorV"}],"title":"serviceUnavailable","url":"\/documentation\/sotocore\/awsservererror\/serviceunavailable","type":"topic","abstract":[],"kind":"symbol"},"doc://codes.soto.soto-core/documentation/SotoCore/AWSServerError/AWSErrorType-Implementations":{"identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/AWSServerError\/AWSErrorType-Implementations","role":"collectionGroup","title":"AWSErrorType Implementations","url":"\/documentation\/sotocore\/awsservererror\/awserrortype-implementations","type":"topic","abstract":[],"kind":"article"},"doc://codes.soto.soto-core/documentation/SotoCore/HeaderDecodingError":{"type":"topic","abstract":[{"type":"text","text":"Decoding error returned when extracting data out of response headers"}],"kind":"symbol","identifier":"doc:\/\/codes.soto.soto-core\/documentation\/SotoCore\/HeaderDecodingError","url":"\/documentation\/sotocore\/headerdecodingerror","fragments":[{"text":"struct","kind":"keyword"},{"kind":"text","text":" "},{"text":"HeaderDecodingError","kind":"identifier"}],"title":"HeaderDecodingError","navigatorTitle":[{"text":"HeaderDecodingError","kind":"identifier"}],"role":"symbol"},"doc://codes.soto.soto-core/s5ErrorP":{"identifier":"doc:\/\/codes.soto.soto-core\/s5ErrorP","type":"unresolvable","title":"Swift.Error"}}}